src/encoded/static/components/cart/analysis.js

```diff
@@ -0,0 +1,207 @@
+
+                    // Add a new compiled analysis object.
+                    return analyses.concat({
+                        title: representativeAnalysis.title,
+                        status: representativeAnalysis.status,
+                        lab: lab ? lab.slice(0, -1) : '', // Trim last matching space
+                        version: representativeAnalysis.pipeline_version || '',
+                        assembly: representativeAnalysis.assembly || '',
+                        annotation: representativeAnalysis.genome_annotation || '',
+                        assemblyAnnotationValue: computeAssemblyAnnotationValue(representativeAnalysis.assembly, representativeAnalysis.genome_annotation),
+                        files: assemblyFiles,
+                        analysisObjects: analysesByTitle[title],
+                    });
+                }
+
+                // Don't add compiled analysis objects with no files.
+                return analyses;
+            }, []);
+        }
+    }
+    return compiledAnalyses;
+};
+
+
+/**
+ * [].sort() callback to compare two compiled-analysis version numbers of the form
+ * "major.minor.patch" with optional "minor" and "patch."
+ * @param {object} aAnalysis First compiled analysis object to compare
+ * @param {object} bAnalysis Second compiled analysis object to compare
+ *
+ * @return {number} [].sort() result to determine ordering
+ */
+const compareAnalysisVersions = (aAnalysis, bAnalysis) => {
+    /**
+     * Break a string version number into a corresponding vector of numbers. Examples:
+     * "0.3.75" => [0, 3, 75]
+     * "10.81" => [10, 81]
+     * "2.0.0" => [2, 0, 0]
+     * @param {string} version Version number in the form "major.minor.patch".
+     *
+     * @return {array} Version number converted to numeric vector
+     */
+    const deconstructVersion = (version) => (
+        version.match(/(\d+)(\.(\d+))*?(\.(\d+))*?/g).map((versionElement) => +versionElement)
+    );
+
+    // If either or both analyses have no version number, sort the empty version number after a
+    // non-empty one.
+    if (aAnalysis.version === '' || bAnalysis.version === '') {
+        if (aAnalysis.version !== bAnalysis.version) {
+            return aAnalysis.version === '' ? 1 : -1;
+        }
+
+        // Both empty version numbers.
+        return 0;
+    }
+
+    // Deconstruct the version numbers so we can compare numerically. This avoids having "10" sort
+    // before "2."
+    const aDeconstructed = deconstructVersion(aAnalysis.version);
+    const bDeconstructed = deconstructVersion(bAnalysis.version);
+
+    // Compare each deconstructed element of the version numbers. Use "-1" for non-existent minor
+    // and patch version elements so they sort before "0."
+    return (
+        (bDeconstructed[0] - aDeconstructed[0])
+            || ((bDeconstructed[1] ?? -1) - (aDeconstructed[1] ?? -1))
```

Comment:
Won't this cause IE11 to fail? We are not yet in August.

Reply:
The babel plugin I added transpiles it back to IE11? I tried it and in IE11 and it worked, and the bundle.js for the lines shows:

```
  return bDeconstructed[0] - aDeconstructed[0] || ((_bDeconstructed$ = bDeconstructed[1]) !== null && _bDeconstructed$ !== void 0 ? _bDeconstructed$ : -1) - ((_aDeconstructed$ = aDeconstructed[1]) !== null && _aDeconstructed$ !== void 0 ? _aDeconstructed$ : -1) || ((_bDeconstructed$2 = bDeconstructed[2]) !== null && _bDeconstructed$2 !== void 0 ? _bDeconstructed$2 : -1) - ((_aDeconstructed$2 = aDeconstructed[2]) !== null && _aDeconstructed$2 !== void 0 ? _aDeconstructed$2 : -1);
```